roles/openstack/nova/tasks/cleanup.yml

```diff
@@ -0,0 +1,24 @@
+---
+# This file is used to cleanup OpenStack nova resources
+
+- name: Release floating IPs
+  os_floating_ip:
+      server: "{{ item.value.server | default(instance_name) }}"
+      state: absent
+  with_dict: "{{ provisioner.floating_ips }}"
+  when: provisioner.floating_ips is defined
```

Comment:
Where can `floating_ips` be defined? seems like we are going to have a leakage here.
Also, Have you verified that "absent" will actually delete the floater? I seem to recall a bug that it would only disconnect.